aten/src/ATen/native/Distributions.cpp

```diff
@@ -209,6 +209,7 @@ Tensor _dirichlet_grad_cpu(const Tensor& x, const Tensor& alpha, const Tensor& t
  */
 
 Tensor _s_poisson_cpu(const Tensor& lambda, Generator *gen) {
+  TORCH_CHECK((lambda > 0).all().item<uint8_t>(), "invalid Poisson rate, expected rates to be positive");
```

Comment:
maybe move this inside the kernel instead?

Reply:
Sure, that sounds like a better idea.